lib/test.js

```diff
@@ -88,6 +89,20 @@ function Test (options) {
   Readable.apply(this, options)
 }
 
+Test.prototype.beforeEach = function (fn) {
+  this.on('beforechild', fn)
```

Comment:
I did a bit more discussion with @bcoe about the use cases for mocha and lab tests.

The simple answer is "no, not at all".  The longer answer is that we probably need to either do this in a completely different way, or just leave it out of node-tap and move it into some other module with the explicit aim to re-implement the mocha API in terms of tap.